src/js/webSerial.js

```diff
@@ -88,21 +87,27 @@ class WebSerial extends EventTarget {
             filters: webSerialDevices,
         });
 
-        this.port_counter = 1;
+        this.portCounter = 1;
         this.ports = ports.map(function (port) {
             return this.createPort(port);
         }, this);
     }
 
     async requestPermissionDevice() {
-        const permissionPort = await navigator.serial.requestPort({
-            filters: webSerialDevices,
-        });
-        const found = this.ports.find(port => port.port === permissionPort);
-        if (!found) {
-            return this.handleNewDevice(permissionPort);
+        let newPermissionPort = null;
+        try {
+            const permissionPort = await navigator.serial.requestPort({
+                filters: webSerialDevices,
+            });
+            const newPort = this.ports.find(port => port.port === permissionPort);
+            if (!newPort) {
+                newPort = this.handleNewDevice(permissionPort);
+            }
+            console.info("User selected device from permissions:", newPort.path);
+        } catch (error) {
+            console.error("User didn't select any device when requesting permission:", error);
         }
-        return null;
+        return newPermissionPort;
```

Comment:
`newPermissionPort` remains `null` ?

Reply:
When the user does not select any port in the request permission yes. It's a way to know if it has selected one or not.

Reply:
Cannot find where `newPermissionPort` is set (only on line 97) ?

Reply:
True, at some change I lost it... good catch, I will fix it asap

Reply:
@haslinghuis fixed!!!